src/BizHawk.Client.Common/movie/tasproj/ZwinderStateManager.cs

```diff
@@ -311,6 +311,49 @@ public void EvictReserved(int frame)
 			}
 		}
 
+		public void Unreserve(int frame)
+		{
+			// Before removing the state, check if it should be still be reserved.
+			// For now, this just means checking if we need to keep this state to satisfy the ancient interval.
+			if (ShouldKeepForAncient(frame))
+				return;
+
+			EvictReserved(frame);
+		}
+
+		private void PossiblyMoveToReserved(ZwinderBuffer buffer, int stateIndex)
+		{
+			var state = buffer.GetState(stateIndex);
+
+			// Add to reserved buffer if externally reserved, or if it matches an "ancient" state consideration
+			if (_reserveCallback(state.Frame) || ShouldKeepForAncient(state.Frame))
+				AddToReserved(state);
+			else
+				StateCache.Remove(state.Frame);
+		}
+
+		/// <summary>
+		/// Will removing the state on this leave us with a gap larger than the ancient interval?
+		/// </summary>
+		private bool ShouldKeepForAncient(int frame)
+		{
+			int index = StateCache.BinarySearch(frame + 1);
+			if (index < 0)
+				index = ~index;
+			if (index == StateCache.Count)
+			{
+				// There is no future state... so why are we wanting to evict this state?
+				// We aren't decaying from _recent, that's for sure. So,
+				return false;
+			}
+			if (index <= 1)
+				return true; // This should never happen.

```

Comment:
While `index < 0` cannot happen, this isn't true for the other possible cases.

`index == 0` will only happen when `frame < 0`, which is arguably nonsensical, so the return value can be anything. 
However `index == 1` can very much happen when `frame = 0`, and returning `true` is correct here. So while the code is fine, the comment is confusing.